pkg/storage/backend-storage/backend-storage.go

```diff
@@ -107,6 +111,152 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
 	return pvcForMigrationTargetFromStore(pvcStore, migration)
 }
 
+func RecoverFromBrokenMigration(client kubecli.KubevirtClient, migration *corev1.VirtualMachineInstanceMigration, pvcStore cache.Store, vmi *corev1.VirtualMachineInstance, launcherImage string) error {
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX.
+		// In both cases we consider the migration as failed.
+		migration.Status.Phase = corev1.MigrationFailed
+		return nil
+	}
+
+	// An interrupted migration exists. Using a job to check if the source PVC contains /meta/migrated,
+	// which would indicate that the libvirt migration finished.
+	// A JobComplete condition indicates the file is present, the migration was successful and the target PVC prevails
+	// A JobFailed condition indicated the file is absent, the migration didn't finish and the source PVC prevails
+
+	jobName := "recover-" + migration.Name
+
+	job, err := client.BatchV1().Jobs(vmi.Namespace).Get(context.Background(), jobName, metav1.GetOptions{})
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		job = buildRecoveryJob(jobName, migration.Status.MigrationState.SourcePersistentStatePVCName, launcherImage)
+		job, err = client.BatchV1().Jobs(vmi.Namespace).Create(context.Background(), job, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Give the job 5 seconds to complete
+	err = virtwait.PollImmediately(time.Second, 5*time.Second, func(ctx context.Context) (done bool, err error) {
+		job, err = client.BatchV1().Jobs(job.Namespace).Get(ctx, job.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+		for _, c := range job.Status.Conditions {
+			switch c.Type {
```

Comment:
There should be a default case as conditions can be added at any point, even by external actors.